rc2/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+        r[0] = r[0].wrapping_add(self.exp_key[*j]).wrapping_add(r[3] & r[2]).wrapping_add(!r[3] & r[1]);
+        *j += 1;
+        r[0] = (r[0] << 1) | (r[0] >> 15);
+
+        r[1] = r[1].wrapping_add(self.exp_key[*j]).wrapping_add(r[0] & r[3]).wrapping_add(!r[0] & r[2]);
+        *j += 1;
+        r[1] = (r[1] << 2) | (r[1] >> 14);
+
+        r[2] = r[2].wrapping_add(self.exp_key[*j]).wrapping_add(r[1] & r[0]).wrapping_add(!r[1] & r[3]);
+        *j += 1;
+        r[2] = (r[2] << 3) | (r[2] >> 13);
+
+        r[3] = r[3].wrapping_add(self.exp_key[*j]).wrapping_add(r[2] & r[1]).wrapping_add(!r[2] & r[0]);
+        *j += 1;
+        r[3] = (r[3] << 5) | (r[3] >> 11);
+    }
+
+    fn mash(&self, r: &mut [u16; 4]) {
+        r[0] = r[0].wrapping_add(self.exp_key[(r[3] & 63) as usize]);
+        r[1] = r[1].wrapping_add(self.exp_key[(r[0] & 63) as usize]);
+        r[2] = r[2].wrapping_add(self.exp_key[(r[1] & 63) as usize]);
+        r[3] = r[3].wrapping_add(self.exp_key[(r[2] & 63) as usize]);
+    }
+
+    fn reverse_mix(&self, r: &mut [u16; 4], j: &mut usize) {
+        r[3] = (r[3] << 11) | (r[3] >> 5);
+        r[3] = r[3].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[2] & r[1]).wrapping_sub(!r[2] & r[0]);
+        *j -= 1;
+
+        r[2] = (r[2] << 13) | (r[2] >> 3);
+        r[2] = r[2].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[1] & r[0]).wrapping_sub(!r[1] & r[3]);
+        *j -= 1;
+
+        r[1] = (r[1] << 14) | (r[1] >> 2);
+        r[1] = r[1].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[0] & r[3]).wrapping_sub(!r[0] & r[2]);
+        *j -= 1;
+
+        r[0] = (r[0] << 15) | (r[0] >> 1);
+        r[0] = r[0].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[3] & r[2]).wrapping_sub(!r[3] & r[1]);
+        *j = j.wrapping_sub(1);
+    }
+
+    fn reverse_mash(&self, r: &mut [u16; 4]) {
+        r[3] = r[3].wrapping_sub(self.exp_key[(r[2] & 63) as usize]);
+        r[2] = r[2].wrapping_sub(self.exp_key[(r[1] & 63) as usize]);
+        r[1] = r[1].wrapping_sub(self.exp_key[(r[0] & 63) as usize]);
+        r[0] = r[0].wrapping_sub(self.exp_key[(r[3] & 63) as usize]);
+    }
+
+    fn encrypt(&self, block: [u8; 8]) -> [u8; 8] {
+        let mut r: [u16; 4] = [0; 4];
+
+        r[0] = ((block[1] as u16) << 8) + (block[0] as u16);
+        r[1] = ((block[3] as u16) << 8) + (block[2] as u16);
+        r[2] = ((block[5] as u16) << 8) + (block[4] as u16);
+        r[3] = ((block[7] as u16) << 8) + (block[6] as u16);
+
+        let mut j = 0;
+
+        for i in 0..16 {
+            self.mix(&mut r, &mut j);
+            if i == 4 || i == 10 {
+                self.mash(&mut r);
+            }
+        }
+
+        let mut ct: [u8; 8] = [0; 8];
+        ct[0] = r[0] as u8;
+        ct[1] = (r[0] >> 8) as u8;
+        ct[2] = r[1] as u8;
+        ct[3] = (r[1] >> 8) as u8;
+        ct[4] = r[2] as u8;
+        ct[5] = (r[2] >> 8) as u8;
+        ct[6] = r[3] as u8;
+        ct[7] = (r[3] >> 8) as u8;
+
+        ct
```

Comment:
You can construct and return this array immediately like this:
```Rust
[
    r[0] as u8,
    (r[0] >> 8) as u8,
    ...
]
```

Same can be done at the lines 107-112 for construction of `r`. Initialization is probably optimized away, so performance wise they are the same, but it will make code a bit shorter.

Later I will probably add `u16` methods to the `byte-tools` crate, so it could be written even shorter.